src/ring_theory/jacobson_ideal.lean

```diff
@@ -86,6 +96,133 @@ theorem mem_jacobson_iff {I : ideal R} {x : R} :
         neg_mul_eq_neg_mul_symm, neg_mul_eq_mul_neg, mul_comm x y]; exact M.mul_mem_right hy)
     (him hz)⟩
 
+theorem eq_jacobson_iff_Inf_maximal {I : ideal R} :
+  I.jacobson = I ↔ ∃ M ⊆ {J : ideal R | J.is_maximal ∨ J = ⊤}, I = Inf M :=
```

Comment:
I think the statement `∃ M : set (ideal R), (∀ J ∈ M, J.is_maximal) ∧ I = Inf M` is nicer without the set notation. It should be equivalent, since the infimum of any set containing `⊤` is the same as the infimum after removing `⊤`.

Reply:
The only reason I \added the `∨ J = ⊤` was that it seems like most uses of this end up being alongside maps and comaps, which often ends up giving the potential of the ideal being top, and it seemed tedious to keep proving that it doesn't matter either way. Not sure if that is worth the trade-off with making this theorem more wordy or not (Either way I'll go ahead and switch away from the set notation and just use a ∀instead)

Reply:
So I guess it depends which direction you're using the lemma in. I'm happy to leave it as it is.

Reply:
I ended up leaving the statement, but I've removed the set notation and added a docstring clarifying why top is allowed to be included in the set `M`